storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -561,8 +561,7 @@ public static void validateRemoteStorageOnlyIfSystemEnabled(Map<?, ?> props, boo
     @SuppressWarnings("unchecked")
     private static void validateRemoteStorageRequiresDeleteCleanupPolicy(Map<?, ?> props) {
         List<String> cleanupPolicy = (List<String>) props.get(TopicConfig.CLEANUP_POLICY_CONFIG);
-        Set<String> policySet = cleanupPolicy.stream().map(policy -> policy.toLowerCase(Locale.getDefault())).collect(Collectors.toSet());
-        if (!Set.of(TopicConfig.CLEANUP_POLICY_DELETE).equals(policySet)) {
+        if (cleanupPolicy.size() != 1 || !TopicConfig.CLEANUP_POLICY_DELETE.equals(cleanupPolicy.get(0))) {
```

Comment:
We need to change the logic to accommodate for empty cleanupPolicy with remote storage.